web/src/message_list.ts

```diff
@@ -66,12 +66,17 @@ export class MessageList {
     last_message_historical?: boolean;
     should_trigger_message_selected_event?: boolean;
 
-    constructor(opts: {
-        data: MessageListData;
-        filter: Filter;
-        excludes_muted_topics: boolean;
-        is_node_test: boolean;
-    }) {
+    constructor(
+        opts: (
+            | {
+                  data: MessageListData;
+              }
+            | {
+                  data?: undefined;
+                  filter: Filter;
+              }
+        ) & {excludes_muted_topics?: boolean; is_node_test?: boolean},
+    ) {
```

Comment:
This kinda feels like an API where we should as a follow-up just make the caller do `new MessageListData` if it doesn't already have one. We have very few code paths that make these....

Reply:
Turns out it was only tests using this shortcut. Putting up a fix, good idea.